projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -700,39 +716,77 @@ public static int WriteShort(Span<byte> span, ushort val)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe int WriteShortstr(Span<byte> span, string val)
+        public static int WriteShortstr(Span<byte> span, string val)
         {
+            if (string.IsNullOrEmpty(val))
```

Comment:
AFAIK strings are not allowed to be null for all public APIs, this would allow it. Do we to change the behaviour?

Reply:
Was more a precaution. Null checks could be added to the public API to prevent them from ever being null here.

Reply:
I'd not add checks, but rather eventually add the nullability annotations to the whole assembly, but this is something for another time =)